### PR TITLE
Update examples in docs to reflect that qreg and qspan are deprecated

### PR DESCRIPTION
**Pull request by bmhowe23**
As a follow up to #983 and #1029, this changes the `qreg` and `qspan` to `qvector`, `qarray`, and `qview` as appropriate. It also change some examples to `literalinclude` (where possible).

### CODE REVIEW

File: docs/sphinx/examples/cpp/algorithms/grover.cpp [hunk 1]
```diff
@@ -9,11 +9,14 @@ __qpu__ void reflect_about_uniform(cudaq::qview<> q) {
...
-  x(q);
-  h(q);
+  // Compute (U) Action (V) produces
+  // U V U::Adjoint
+  cudaq::compute_action(
+      [&]() {
+        h(q);
+        x(q);
+      },
+      [&]() { z<cudaq::ctrl>(ctrlQubits, lastQubit); });
```
Comment (schweitzpgi) [inline docs/sphinx/examples/cpp/algorithms/grover.cpp:19]: Nice

**Review by schweitzpgi — APPROVED** [review]
Skimmed some of this, but LGTM.

File: .github/workflows/config/spelling_allowlist.txt [hunk 2]
```diff
@@ -10,9 +10,11 @@ CPTP
 CPU
 CPUs
 CUDA
+Composability
```
Comment (bettinaheim) [inline .github/workflows/config/spelling_allowlist.txt:13]: I think you can probably lowercase it in the allow list and the spellcheck should be smart enough to recognize capitalization at the beginning of a sentence.

**Review by bettinaheim — APPROVED** [review]
Thanks!

File: .github/workflows/config/spelling_allowlist.txt (hunk 2 above)
Comment (bmhowe23) [reply .github/workflows/config/spelling_allowlist.txt]: Thanks. Updated in [a41da94](https://github.com/NVIDIA/cuda-quantum/pull/1040/commits/a41da9475b844ed6fca05fd907469c41bb65ffa4).

**Pull request merged**